class Point(our x, our y)

async fn main() {
    our p = Point(22, 44)
    # Test that we execute `test(p)` (and hence see its output)
    # before we detect the error here
    test(p).await.x += 1 #! RUN ERROR cannot write to shared fields
    print(p).await
}

async fn test(shleased p) -> {
    print("Hi").await #! OUTPUT Hi
    p
}